superset-frontend/src/components/Button/index.tsx

```diff
@@ -52,264 +45,164 @@ export interface ButtonProps {
-          mix(0.1, theme.colors.grayscale.light5, theme.colors.info.dark1)};
-        color: ${({ theme }) => theme.colors.grayscale.light5};
-      }
-      &:active {
-        background-color: ${({ theme }) =>
-          mix(0.2, theme.colors.grayscale.dark2, theme.colors.info.dark1)};
-        color: ${({ theme }) => theme.colors.grayscale.light5};
-      }
-    }
-    &[disabled],
-    &[disabled]:hover {
-      background-color: ${({ theme }) => theme.colors.grayscale.light2};
-      color: ${({ theme }) => theme.colors.grayscale.base};
-    }
-  }
+  let backgroundColor = primary.light4;
+  let backgroundColorHover = mix(0.1, primary.base, primary.light4);
+  let backgroundColorActive = mix(0.25, primary.base, primary.light4);
+  let backgroundColorDisabled = grayscale.light2;
+  let color = primary.dark1;
+  let colorHover = color;
+  let borderWidth = 0;
+  let borderStyle = 'none';
+  let borderColor = 'transparent';
+  let borderColorHover = 'transparent';
+  let borderColorDisabled = 'transparent';
 
-  /* big Call to Action buttons */
-  &.cta {
-    min-width: ${({ theme }) => theme.gridUnit * 36}px;
-    min-height: ${({ theme }) => theme.gridUnit * 8}px;
-    text-transform: uppercase;
+  if (buttonStyle === 'primary') {
+    backgroundColor = primary.dark1;
+    backgroundColorHover = mix(0.1, grayscale.light5, primary.dark1);
+    backgroundColorActive = mix(0.2, grayscale.dark2, primary.dark1);
+    color = grayscale.light5;
+    colorHover = color;
+  } else if (buttonStyle === 'tertiary' || buttonStyle === 'dashed') {
+    backgroundColor = grayscale.light5;
+    backgroundColorHover = grayscale.light5;
+    backgroundColorActive = grayscale.light5;
+    backgroundColorDisabled = grayscale.light5;
+    borderWidth = 1;
+    borderStyle = buttonStyle === 'dashed' ? 'dashed' : 'solid';
+    borderColor = primary.dark1;
+    borderColorHover = primary.light1;
+    borderColorDisabled = grayscale.light2;
+  } else if (buttonStyle === 'danger') {
+    backgroundColor = error.base;
+    backgroundColorHover = mix(0.1, grayscale.light5, error.base);
+    backgroundColorActive = mix(0.2, grayscale.dark2, error.base);
+    color = grayscale.light5;
+    colorHover = color;
+  } else if (buttonStyle === 'warning') {
+    backgroundColor = warning.base;
+    backgroundColorHover = mix(0.1, grayscale.dark2, warning.base);
+    backgroundColorActive = mix(0.2, grayscale.dark2, warning.base);
+    color = grayscale.light5;
+    colorHover = color;
+  } else if (buttonStyle === 'success') {
+    backgroundColor = success.base;
+    backgroundColorHover = mix(0.1, grayscale.light5, success.base);
+    backgroundColorActive = mix(0.2, grayscale.dark2, success.base);
+    color = grayscale.light5;
+    colorHover = color;
+  } else if (buttonStyle === 'link') {
+    backgroundColor = 'transparent';
+    backgroundColorHover = 'transparent';
+    backgroundColorActive = 'transparent';
+    colorHover = primary.base;
```

Comment:
Nice cleanup!

Reply:
Thanks! I think it's easier to read now.